UPGRADING.md

```diff
@@ -0,0 +1,164 @@
+
+### Coordinated Consensus Rules Changes
+
+- Using the [extension system](https://docs.kwil.com/docs/extensions/overview), Kwil now has a system for nodes to coordinate changing consensus rules at a specific block height. This allows for a more flexible upgrades, such as changing the block size or adding new extensions. Previously, these changes could only be made via launching an entirely new network. - **[Learn more](https://docs.kwil.com/docs/extensions/fork-background)**
+- Note: Consensus rule changes are still experimental and may change depending on feedback.
+
+### JSON-RPC Server
+
+- Kwil now includes a JSON-RPC server that allows for querying the Kwil node using JSON-RPC.
+- The JSON-RPC listen address can be set with `jsonrpc_listen_addr` in the node's `config.toml` file. By default, the JSON-RPC server listens on port 8484.
+- Kwil CLI, GO SDK, and JS SDK have been updated to use the JSON-RPC server.
+
+#### Breaking Changes:
+
+- The HTTP server (`http_listen_addr`) is DEPRECATED and will be removed in the susbequent release.
+- The gRPC server (`grpc_listen_addr`) is no longer exposed.
+
+### Node Configuration Changes
+
+In the node's `config.toml` file, the following changes have been made:
+
+- Added `app.rpc_timeout`, which imposes a timeout on RPC requests. The default is 45 seconds.
+- Added `app.db_read_timeout`, which imposes a timout on read-only DB transactions (i.e. actions/procedures with a [`view` tag](https://docs.kwil.com/docs/kuneiform/supported-features#access-tags) or an ad-hoc SELECT query). The default 5 seconds.
+- Added `rpc.broadcast_tx_timout`, which imposes a timeout for awaiting transaction confirmation when using the `--sync` flag in `kwil-cli`. The default is 10 seconds.
+- Added `app.snapshots` to support snapshot creation. See the `StateSyncConfig` struct for more information.
+- Added `app.snapshot_file` to suport starting a network with a state at genesis. This is to support the Network Migrations feature.
+
+### CLI Changes
+
+- The `kwil-cli` has been updated to use the JSON-RPC server.
+- The `--kwil-provider` flag is DEPRECATED and will be removed in the subsequent release. Use the `--provider` flag instead.
+- Added `kwil-cli utils parse` to parse a Kuneiform file and output the schema as JSON.
+- Added `kwil-cli utils decode-tx` to decode a transaction from a base64 string.
+
+#### Breaking Changes
+
+- Because Kwil-CLI now communicates over JSON-RPC, you should pass the JSON-RPC listen address to the CLI with the `--provider` flag or `kwil-cli configure` command.
+- For uncomfirmed transactions, `kwild` now returns `null` for the `tx` field. Using older versions of `kwil-cli` to query unconfirmed transactions may have unexpected behavior.
```

Comment:
Maybe specify that the query-tx api returns no tx data for unconfirmed transactions.

Reply:
I think that is what: "- For uncomfirmed transactions, `kwild` now returns `null` for the `tx` field. Using older versions of `kwil-cli` to query unconfirmed transactions may have unexpected behavior." Is that not clear?